indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerConfig.java

```diff
@@ -115,6 +118,13 @@ public void configure(Binder binder)
     INDEX_IO = injector.getInstance(IndexIO.class);
     INDEX_MERGER = injector.getInstance(IndexMerger.class);
     INDEX_MERGER_V9 = injector.getInstance(IndexMergerV9.class);
+    LIFECYCLE = injector.getInstance(Lifecycle.class);
+    try {
+      LIFECYCLE.start();
```

Comment:
@b-slim, I'm trying to understand where this is needed; is this necessary for the druid side of a hadoop task (the part that runs on a task peon) or is this for the hadoop side (the part that runs on map/reduce)? Or both?

I think this really shouldn't be relied on. Consider the case where we are reading from kerberized HDFS but we have S3 deep storage. In that case the lifecycle won't have anything interesting in it (no hdfs module adding anything). So if we need to do an authenticate, it should be done some other way.

For the Druid side I would have guessed this should happen as part of JobHelper.ensurePaths (since you added authenticate there).

For the Hadoop side, I'm not sure, maybe Hadoop would do it automatically? If not, we could add something to "init" of our mappers and reducers.